news/9086.feature.rst

```diff
@@ -0,0 +1,2 @@
+Use git's partial clone feature to speed up pip
+when a git-rev is specified.
```

Comment:
```suggestion
When a revision is specified in a Git URL, use partial clone to speed up source retrieval.
```

This is pip’s news fragment, so of course we are speeding up pip. Let’s be a bit more specific 🙂